pmc_heu.cpp

```diff
@@ -90,8 +90,10 @@ int pmc_heu::search_bounds(const pmc_graph& G, std::vector<int>& C_max) {
         const int v = (*order)[i];
 
         int mc_prev, mc_cur;
-        #pragma omp atomic read acquire
-        mc_prev = mc_cur = mc;
```

Comment:
Is it not possible to keep this? I feel like it should be fine to read as atomic since we have the critical update below anyways (this was the main problem if I understand it correctly).

Reply:
I think it'd better be locked in the same manner as the update below. i.e., `mc` is read and written inside the critical section. I googled whether it's safe to mix-use atomic and mutex but couldn't find any reference though.